cpu/sam0_common/periph/rtt.c

```diff
@@ -56,48 +56,30 @@ static inline void _rtt_reset(void)
 #ifdef CPU_SAMD21
 static void _rtt_clock_setup(void)
 {
-    /* RTC uses External 32,768KHz Oscillator because OSC32K isn't accurate
-     * enough (p1075/1138). Also keep running in standby. */
-    SYSCTRL->XOSC32K.reg =  SYSCTRL_XOSC32K_ONDEMAND |
-                            SYSCTRL_XOSC32K_EN32K |
-                            SYSCTRL_XOSC32K_XTALEN |
-                            SYSCTRL_XOSC32K_STARTUP(6) |
-#if RTT_RUNSTDBY
-                            SYSCTRL_XOSC32K_RUNSTDBY |
-#endif
-                            SYSCTRL_XOSC32K_ENABLE;
-
-    /* Setup clock GCLK2 with divider 1 */
-    GCLK->GENDIV.reg = GCLK_GENDIV_ID(2) | GCLK_GENDIV_DIV(1);
-    while (GCLK->STATUS.bit.SYNCBUSY) {}
-
-    /* Enable GCLK2 with XOSC32K as source. Use divider without modification
-     * and keep running in standby. */
-    GCLK->GENCTRL.reg = GCLK_GENCTRL_ID(2) |
-                        GCLK_GENCTRL_GENEN |
-#if RTT_RUNSTDBY
-                        GCLK_GENCTRL_RUNSTDBY |
-#endif
-                        GCLK_GENCTRL_SRC_XOSC32K;
-    while (GCLK->STATUS.bit.SYNCBUSY) {}
-
-    /* Connect GCLK2 to RTC */
-    GCLK->CLKCTRL.reg = GCLK_CLKCTRL_GEN_GCLK2 |
-                        GCLK_CLKCTRL_CLKEN |
-                        GCLK_CLKCTRL_ID(RTC_GCLK_ID);
+    /* Setup clock GCLK2 with OSC32K */
+    GCLK->CLKCTRL.reg = GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN(2) | GCLK_CLKCTRL_ID_RTC;
     while (GCLK->STATUS.bit.SYNCBUSY) {}
 }
-/* !CPU_SAMD21 */
 #else
 static void _rtt_clock_setup(void)
 {
-    /* Turn on power manager for RTC */
-    MCLK->APBAMASK.reg |= MCLK_APBAMASK_OSC32KCTRL;
-
-    /* set clock source */
+    /* RTC source clock is external oscillator at 32kHz */
+#if EXTERNAL_OSC32_SOURCE
+    OSC32KCTRL->XOSC32K.bit.EN1K = 1; /* needed at least on SAME54 */
```

Comment:
Should be EN32K.

Reply:
Changed it.
I thought I had to do this to make it work last week and was profoundly confused by it, but now it works without it too, which may or may not reassure me of my sanity.